core/templates/dev/head/player/InteractionService.js

```diff
@@ -0,0 +1,94 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Shared utilities for interactions.
+ *
+ * @author wxyxinyu@gmail.com (Xinyu Wu)
+ */
+
+oppia.constant('DEFAULT_ANSWER_GROUP_CLASSIFICATION_THRESHOLD', 0.3);
+
+oppia.factory('interactionService', [
+    'warningsData', 'DEFAULT_ANSWER_GROUP_CLASSIFICATION_THRESHOLD', function(
+    warningsData, DEFAULT_ANSWER_GROUP_CLASSIFICATION_THRESHOLD) {
+  var stringifyClassifiedRule = function(rule) {
+    var paramList = [];
+    for (var key in rule.inputs) {
+      paramList.push(rule.inputs[key]);
+    }
+    var stringifiedRule = rule.rule_type;
+    stringifiedRule += '(' + paramList.join(',') + ')';
+    return stringifiedRule;
+  };
+  return {
+    'classifyAnswer': function(
+      answer, answerClassificationData, RULES) {
```

Comment:
Consider indenting this by two more to distinguish from the next line.

Also, RULES should not be capitalized.

Also, since this is such a crucial method, I would suggest adding jsdoc or similar to explain what this method does and what the args are.